### PR TITLE
ncm-metaconfig: Include Singularity service

### PR DESCRIPTION
**Pull request by alvarosimon**
Include Singularity configuration support in Quattor metaconfig
- Fix: #953

### CODE REVIEW

File: ncm-metaconfig/src/main/metaconfig/singularity/pan/schema.pan [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+@documentation{
+singularity.conf settings
+This is the global configuration file for Singularity. This file controls
+what the container is allowed to do on a particular host, and as a result
+this file must be owned by root.
+}
+type service_singularity = {
+    include type_singularity_allow
+    include type_singularity_user_config
```
Comment (stdweird) [inline ncm-metaconfig/src/main/metaconfig/singularity/pan/schema.pan]: use
```pan
'config' ? type_singularity_user_config
```
and remove the `config-` from `type_singularity_user_config`

**Comment by alvarosimon** [comment]
@stdweird something like this? I don't know if the type names are fine or a bit confusing...

**Comment by alvarosimon** [comment]
@stdweird I did a rebase with master

**Comment by stdweird** [comment]
can you also squash all your commits in one, and start it with `ncm-metaconfig:`

File: ncm-metaconfig/src/main/metaconfig/singularity/tests/profiles/config.pan [hunk 2]
```diff
@@ -0,0 +1,7 @@
+object template config;
+
+include 'metaconfig/singularity/config';
+
+prefix "/software/components/metaconfig/services/{/etc/singularity/singularity.conf}/contents";
+"enable/overlay" = true;
+"bind/path" = list("/my/scratch", "/etc/singularity/default-nsswitch.conf:/etc/nsswitch.conf");
```
Comment (stdweird) [inline ncm-metaconfig/src/main/metaconfig/singularity/tests/profiles/config.pan]: this should be allowed in the schema

File: ncm-metaconfig/src/main/metaconfig/singularity/pan/schema.pan [hunk 3]
```diff
@@ -0,0 +1,118 @@
+declaration template metaconfig/singularity/schema;
+
+include 'pan/types';
+
+type directory = string with match(SELF,'[^/]+/?$');
```
Comment (stdweird) [inline ncm-metaconfig/src/main/metaconfig/singularity/pan/schema.pan]: do not define these generic types here. should be in `template-library-core` or just use the type `absolute_file_path` instead

File: ncm-metaconfig/src/main/metaconfig/singularity/pan/schema.pan [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+type type_singularity_allow_ns = {
```
Comment (stdweird) [inline ncm-metaconfig/src/main/metaconfig/singularity/pan/schema.pan]: no need to prefix the type names with `type_`

File: ncm-metaconfig/src/main/metaconfig/singularity/pan/schema.pan [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    show up in the container}
+    'slave' : boolean = true
+} = dict();
+
+type type_singularity_bind_path = {
+    @{Define a list of files/directories that should be made available from within
+    the container. The file or directory must exist within the container on
+    which to attach to. you can specify a different source and destination
+    path (respectively) with a colon; otherwise source and dest are the same}
+    'path' ? string[]
```
Comment (stdweird) [inline ncm-metaconfig/src/main/metaconfig/singularity/pan/schema.pan]: you need to make a type here that allows `/abc:/def`; so absolute filename forllowed by optional absolute filename

**Comment by alvarosimon** [comment]
@stdweird something like this?

**Comment by stdweird** [comment]
retest this please

**Comment by stdweird** [comment]
LGTM

**Pull request merged**